packages/pigeon/lib/swift_generator.dart

```diff
@@ -608,9 +608,13 @@ import FlutterMacOS
         // Special-cased to avoid warnings about using 'as' with Any.
         return value;
       } else if (type.baseName == 'int') {
-        final String orString =
-            type.isNullable ? 'nilOrValue($value)' : '$value as! Int64';
-        return '($value is Int32) ? Int64($value as! Int32) : $orString';
+        if (type.isNullable) {
+          // Nullable ints need to check for NSNull, nil, and Int32 before casting can be done safely.
```

Comment:
Nit: we're not checking for nil, it's handled automatically in the simple Int64? cast (modulo the bug).